tools/sdmx_import/sdmx_cli.py

```diff
@@ -0,0 +1,283 @@
+            missing_flags.append(f"--{flag_name}")
+
+    if missing_flags:
+        raise ValueError(
+            f"Missing required flags for {command}: {', '.join(missing_flags)}")
+
+
+def setup_logging() -> None:
+    """
+    Configure logging based on verbosity flags from FLAGS.
+    """
+    if FLAGS.quiet:
+        level = logging.ERROR
+    elif FLAGS.verbose:
+        level = logging.DEBUG
+    else:
+        level = logging.INFO
+
+    logging.set_verbosity(level)
+    logging.set_stderrthreshold(level)
+
+
+def main(argv) -> None:
+    """Main entry point for the CLI tool."""
+    # Command has already been extracted from argv
+    if len(argv) > 1:
+        raise ValueError(f"Unexpected arguments: {argv[1:]}")
+
+    # Setup logging
+    setup_logging()
+
+    # Validate command was set
+    if _COMMAND is None:
+        raise ValueError(
+            f"No command specified. Use: {', '.join(_COMMAND_HANDLERS.keys())}")
+
+    # Validate required flags for the command
+    validate_required_flags_for_command(_COMMAND)
+
+    # Route to appropriate handler using the mapping
+    if _COMMAND in _COMMAND_HANDLERS:
+        _COMMAND_HANDLERS[_COMMAND]()
+    else:
+        raise ValueError(f"Unknown command: {_COMMAND}")
+
+
+def extract_command_from_argv() -> Optional[str]:
+    """
+    Extract command from sys.argv before absl processes it.
+    Removes command from argv and returns it.
+
+    Returns:
+        The command string if valid, None if validation fails
+    """
+    if len(sys.argv) < 2:
+        usage_msg = (
+            "Usage: sdmx_cli.py <command> [flags]\n"
+            f"Commands: {', '.join(_COMMAND_HANDLERS.keys())}\n\n"
+            "Examples:\n"
+            "  # Download metadata\n"
+            "  sdmx_cli.py download-metadata \\\n"
+            "    --endpoint=https://sdmx.oecd.org/public/rest/ \\\n"
+            "    --agency=OECD.SDD.NAD \\\n"
+            "    --dataflow=DSD_NAMAIN1@DF_QNA_EXPENDITURE_GROWTH_OECD \\\n"
+            "    --output_path=metadata.xml\n\n"
+            "  # Download data with filters\n"
+            "  sdmx_cli.py download-data \\\n"
+            "    --endpoint=https://sdmx.oecd.org/public/rest/ \\\n"
+            "    --agency=OECD.SDD.NAD \\\n"
+            "    --dataflow=DSD_NAMAIN1@DF_QNA_EXPENDITURE_GROWTH_OECD \\\n"
+            "    --key=FREQ:Q --key=REF_AREA:USA \\\n"
+            "    --param=startPeriod:2022 \\\n"
+            "    --output_path=data.csv")
+        print(usage_msg)
+        return None
+
+    command = sys.argv[1]
+
+    if command not in _COMMAND_HANDLERS:
+        print(f"Unknown command '{command}'. "
+              f"Valid commands: {', '.join(_COMMAND_HANDLERS.keys())}")
+        return None
+
+    # Remove command from argv so absl.flags doesn't see it
+    sys.argv.pop(1)
+    return command
+
+
+if __name__ == "__main__":
+    # Extract command before absl.app.run() processes argv
+    _COMMAND = extract_command_from_argv()
```

Comment:
Can command parsing be moved within main() to remove need for a global variable for __COMMAND